Firmware/ultralcd.cpp

```diff
@@ -365,7 +365,7 @@ void lcdui_print_percent_done(void)
 		const int8_t nextSheet = eeprom_next_initialized_sheet(sheetNR);
 		if ((nextSheet >= 0) && (sheetNR != nextSheet))
 		{
-			char sheet[sizeof(Sheet::name)];
+			char sheet[sizeof(Sheet::name)+1UL];
```

Comment:
Idea: Can we use `SheetName` struct? That is, replace `char sheet[sizeof(Sheet::name)+1UL];` with `SheetName sheet;`? That would make the initialization more consistent and avoid off by one error.

Then instead of:

```c
eeprom_read_block(sheet, EEPROM_Sheets_base->s[sheetNR].name, sizeof(Sheet::name));
sheet[sizeof(Sheet::name)] = '\0';
```

You would use:

```c
eeprom_read_block(sheet.c, EEPROM_Sheets_base->s[sheetNR].name, sizeof(sheets.c));
sheet.c[sizeof(sheets.c)] = '\0';
```

Reply:
Changing it to
```
			SheetName sheet;
            eeprom_read_block(sheet.c, EEPROM_Sheets_base->s[sheetNR].name, sizeof(sheet.c));
            sheet.c[sizeof(sheet.c)] = '\0';
```
results in more ressource usage: from `248796` to `248820` bytes